includes/class-wc-privacy-exporters.php

```diff
@@ -348,4 +348,49 @@ protected static function get_download_personal_data( $download ) {
 
 		return $personal_data;
 	}
+
+	/**
+	 * Finds and exports customer tokens by email address.
+	 *
+	 * @since 3.4.0
+	 * @param string $email_address The user email address.
+	 * @param int    $page  Page.
+	 * @return array An array of personal data in name value pairs
+	 */
+	public static function customer_tokens_exporter( $email_address, $page ) {
+		$user           = get_user_by( 'email', $email_address ); // Check if user has an ID in the DB to load stored personal data.
+		$data_to_export = array();
+
+		if ( ! $user instanceof WP_User ) {
+			return array(
+				'data' => $data_to_export,
+				'done' => true,
+			);
+		}
+
+		$tokens = WC_Payment_Tokens::get_tokens( array(
+			'user_id' => $user->ID,
+			'limit'   => 10,
+			'page'    => $page,
+		) );
+
+		if ( 0 < count( $tokens ) ) {
+			foreach ( $tokens as $token ) {
+				$data_to_export[] = array(
+					'group_id'    => 'woocommerce_tokens',
+					'group_label' => __( 'Tokens', 'woocommerce' ),
+					'item_id'     => 'token-' . $token->get_id(),
+					'data'        => json_encode( $token->get_data() ),
```

Comment:
Any thoughts on this?

Reply:
I think this is too broad - can we select some important data and mask it where possible? A secure token shouldn't be passed around to transferred to someone else. It's specific to the store owners account.

Reply:
That is a bit tricky. So far we have 2 classes that extend from `WC_Payment_Tokens` and since there are no getters in the base class, there is no good way to get specific data.

For example, `WC_Payment_Token_ECheck` only has a single getter `get_last4`, but `WC_Payment_Token_CC` has a bunch more.